rs/rosetta-api/icrc1/ledger/sm-tests/src/lib.rs

```diff
@@ -3980,6 +3999,27 @@ test_bytes";
         "Expected: {}, got: {}",
         expected_message, message
     );
+
+    args.arg = Encode!(&ApproveArgs {
+        from_subaccount: None,
+        ..approve_args.clone()
+    })
+    .unwrap();
+    args.user_preferences.metadata.utc_offset_minutes = None;
+
+    let message = extract_icrc21_message_string(
+        &icrc21_consent_message(env, canister_id, Principal::anonymous(), args.clone())
+            .unwrap()
+            .consent_message,
+    );
+
+    let expected_message = expected_approve_message.replace("\n\n**Your account:**\nd2zjj-uyaaa-aaaaa-aaaap-4ai-qmfzyha.101010101010101010101010101010101010101010101010101010101010101","\n\n**Your Subaccount:**\n0000000000000000000000000000000000000000000000000000000000000000" )
```

Comment:
Is the way this should be displayed part of the standard, or some ICRC ledger documentation, i.e., that for the anonymous principal, display "Your Subaccount" instead of "Your account"? I couldn't find any examples [here](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/ICRC-21/icrc_21_consent_msg.md).

Also, should the capitalization for "Your account" vs. "Your Subaccount" be unified? (This isn't changed as part of this PR, but I didn't spot it before.)

Reply:
> Is the way this should be displayed part of the standard, or some ICRC ledger documentation, i.e., that for the anonymous principal, display "Your Subaccount" instead of "Your account"? I couldn't find any examples here.

The reason for that is that if the `caller` of the consent message request is anonymous, we cannot know the full account (since the `caller` of the future transaction is not known).

>Also, should the capitalization for "Your account" vs. "Your Subaccount" be unified? (This isn't changed as part of this PR, but I didn't spot it before.)

Yes, this should definitely be made consistent.

Reply:
That makes sense, thanks for the clarification!

Reply:
Done